m3/reporter.go

```diff
@@ -692,9 +692,16 @@ func (r *reporter) reportInternalMetrics() {
 }
 
 func (r *reporter) timeLoop() {
+	t := time.NewTicker(_timeResolution)
+	defer t.Stop()
 	for !r.done.Load() {
 		r.now.Store(time.Now().UnixNano())
-		time.Sleep(_timeResolution)
+		select {
+		case <-t.C:
+			break
```

Comment:
nit: `return`

Reply:
I really need to continue looping in loop and "breaking" out of loop in case of donech being closed

Reply:
Oops, yep. Thanks!